File: src/algorithms/strings/caesar-cipher-encryptor/solution2.py
# O(n) time | O(n) space
def caesarCipherEncryptor(string, key):
	newLetters = []
	newKey = key % 26
	alphabet = list("abcdefghikjlmnopqrstuvwxyz")
	for letter in string:
		newLetters.append(getNewLetter(letter, newKey, alphabet))
	return "".join(newLetters)

def getNewLetter(letter, key, alphabet):
	newLetterCode = alphabet.index(letter) + key
	return alphabet[newLetterCode] if newLetterCode <= 25 else alphabet[-1 + newLetterCode % 25]

File: src/algorithms/strings/caesar-cipher-encryptor/solution2.ts
// O(n) time | O(n) space
export function caesarCipherEncryptor(string: string, key: number) {
  const newLetters = [];
  const newKey = key % 26;
  const alphabet = "abcdefghijklmnopqrstuvwxyz".split("");
  for (const letter of string) {
    newLetters.push(getNewLetter(letter, newKey, alphabet));
  }
  return newLetters.join("");
}

function getNewLetter(letter: string, key: number, alphabet: string[]) {
  const newLetterCode = alphabet.indexOf(letter) + key;
  return alphabet[newLetterCode % 26];
}
